File: quize next-gen-ui/variant-a/fix_emojis.py
import re

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Regex to match the old em-content zobj URLs
    # Example: https://em-content.zobj.net/source/apple/354/hourglass-done_231b.png
    # Pattern: https://em-content\.zobj\.net/source/apple/354/[a-z0-9\-]+_([0-9a-f\-]+)\.png
    pattern = r'https://em-content\.zobj\.net/source/apple/354/[a-z0-9\-]+_([0-9a-f\-]+)\.png'
    replacement = r'https://unpkg.com/emoji-datasource-apple@15.0.1/img/apple/64/\1.png'
    
    new_content = re.sub(pattern, replacement, content)
    
    if content != new_content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated {filepath}")
    else:
        print(f"No changes in {filepath}")

process_file('quiz.js')
process_file('quiz.html')

File: quiz-next-gen-ui/variant-a/fix_emojis.js
const fs = require('fs');
const path = require('path');

function processFile(filepath) {
    const fullPath = path.join(__dirname, filepath);
    let content = fs.readFileSync(fullPath, 'utf8');

    // Regex to match the old em-content zobj URLs
    const pattern = /https:\/\/em-content\.zobj\.net\/source\/apple\/354\/[a-z0-9\-]+_([0-9a-f\-]+)\.png/g;
    const replacement = 'https://unpkg.com/emoji-datasource-apple@15.0.1/img/apple/64/$1.png';
    
    let newContent = content.replace(pattern, replacement);
    
    if (content !== newContent) {
        fs.writeFileSync(fullPath, newContent, 'utf8');
        console.log(`Updated ${filepath}`);
    } else {
        console.log(`No changes in ${filepath}`);
    }
}

processFile('quiz.js');
processFile('quiz.html');
